src/input/tablet_tool.c

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#include <assert.h>
+#include <stdlib.h>
+#include <wlr/types/wlr_tablet_pad.h>
+#include <wlr/types/wlr_tablet_tool.h>
+#include <wlr/util/log.h>
+#include "common/macros.h"
+#include "common/mem.h"
+#include "config/rcxml.h"
+#include "input/cursor.h"
+#include "input/tablet_tool.h"
+#include "input/tablet.h"
+#include "labwc.h"
+
+struct drawing_tablet_tool *
+tablet_tool_from_wlr_tool(struct drawing_tablet *tablet,
+	struct wlr_tablet_tool *wlr_tablet_tool)
+{
+	struct drawing_tablet_tool *tool;
+	wl_list_for_each(tool, &tablet->tablet_tools, link) {
+		if (tool->base == wlr_tablet_tool) {
+			return tool;
+		}
+	}
+
+	return NULL;
+}
+
+static void
+handle_surface_destroy(struct wl_listener *listener, void *data)
+{
+	struct drawing_tablet_tool *tool =
+		wl_container_of(listener, tool, handlers.surface_destroy);
+
+	cursor_set(tool->tablet->seat, LAB_CURSOR_DEFAULT);
+	cursor_update_image(tool->tablet->seat);
+}
+
+static void
+handle_set_cursor(struct wl_listener *listener, void *data)
+{
+	struct drawing_tablet_tool *tool =
+		wl_container_of(listener, tool, handlers.set_cursor);
+	struct wlr_tablet_v2_event_cursor *ev = data;
+
+	struct seat *seat = tool->tablet->seat;
+
+	if (seat->server->input_mode != LAB_INPUT_STATE_PASSTHROUGH) {
+		return;
+	}
+
+	wl_list_remove(&tool->handlers.surface_destroy.link);
```

Comment:
Do I need to guard this against concurrent calls from multiple client?

Reply:
I don't know, but it feels like only the 'focused' client should be allowed to request `set_cursor` (just guessing though).